workers/unity/Packages/com.improbable.gdk.transformsynchronization/Systems/InitializeEntitiesSystem.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Generated.Improbable;
+using Generated.Improbable.Transform;
+using Improbable.Gdk.Core;
+using Unity.Collections;
+using Unity.Entities;
+
+namespace Improbable.Gdk.TransformSynchronization
+{
+    [DisableAutoCreation]
+    [UpdateAfter(typeof(SpatialOSReceiveGroup))]
+    public class InitializeEntitiesSystem : ComponentSystem
+    {
+        private struct Data
+        {
+            [ReadOnly] public readonly int Length;
+            [ReadOnly] public EntityArray Entity;
+            [ReadOnly] public ComponentDataArray<TransformInternal.Component> Transform;
+
+            // This can't work once we put proper interest in. Will need to completely decouple transform and position
+            [ReadOnly] public ComponentDataArray<Position.Component> Position;
+            [ReadOnly] public ComponentDataArray<NewlyAddedSpatialOSEntity> DenotesNewEntity;
+        }
+
+        [Inject] private Data data;
+        [Inject] private WorkerSystem worker;
+
+        protected override void OnUpdate()
+        {
+            for (int i = 0; i < data.Length; ++i)
+            {
+                // Need to split this out into server / client / whatever
+                var transform = data.Transform[i];
+                var defaultToSet = new TransformToSet
+                {
+                    Position = transform.Location.ToUnityVector3() + worker.Origin,
+                    Velocity = transform.Velocity.ToUnityVector3(),
+                    Orientation = transform.Rotation.ToUnityQuaternion()
+                };
+                var defaultToSend = new TransformToSend
+                {
+                    Position = transform.Location.ToUnityVector3() - worker.Origin,
+                    Velocity = transform.Velocity.ToUnityVector3(),
+                    Orientation = transform.Rotation.ToUnityQuaternion()
+                };
+                var previousTransform = new DefferedUpdateTransform
+                {
+                    Transform = transform
+                };
+                var ticksSinceLastUpdate = new TicksSinceLastUpdate
```

Comment:
Does this really need its own component?

Reply:
Yes

Reply:
To give a real answer - I'll check if the component is still needed at all but it does something different from the other components.